EIPS/eip-00.md

```diff
@@ -0,0 +1,828 @@
+| `CALLER`    | `msg.sender`        | The `AA_ENTRY_POINT` address. `AA_SENDER_CREATOR` for the "deployment frame". |
+| `ORIGIN`    | `tx.origin`         | The transaction `sender` address                                              |
+| `CALLDATA*` | `msg.data`          | The transaction data is set to inputs of the corresponding frame              |
+
+#### Nonce validation frame
+
+The `NonceManager` is invoked with the following data:
+
+```solidity
+abi.encodePacked(sender, nonce)
+```
+
+#### Sender deployment frame
+
+The `deployer` address is invoked with the `deployerData[20:]` as call data input.
+It is important that the `deployer` is **not** invoked from the `AA_ENTRY_POINT` but from the `AA_SENDER_CREATOR`.
+This is necessary to guarantee that `AA_ENTRY_POINT` may never initiate a call to a `sender` execution function
+without first completing a successful validation.
+
+The gas limit of this frame is set to `validationGasLimit`.
+The amount of gas used by this frame is referred to as `senderCreationGasUsed`.
+
+The sender deployment frame MUST result in the `sender` address becoming
+initialized with contract code.
+
+#### Sender validation frame
+
+We define the following Solidity struct to represent the AA transaction on-chain:
+
+```solidity
+
+struct TransactionType4 {
+    address sender;
+    uint256 nonce;
+    uint256 validationGasLimit;
+    uint256 paymasterGasLimit;
+    uint256 callGasLimit;
+    uint256 maxFeePerGas;
+    uint256 maxPriorityFeePerGas;
+    uint256 builderFee;
+    bytes paymasterData;
+    bytes deployerData;
+    bytes callData;
+    bytes signature;
+}
+
+```
+
+We then define the following Solidity method and the `sender` of the transaction is invoked with the corresponding data:
+
+```solidity
+
+function validateTransaction(uint256 version, bytes32 txHash, bytes transaction) external returns (uint256 validationData);
+
+```
+
+The gas limit of this frame is set to `validationGasLimit - senderCreationGasUsed`.\
+The `transaction` parameter is interpreted as an ABI encoding of `TransactionType4`.\
+The `txHash` parameter represents the hash of the AA_TX_TYPE transaction with empty signature, as defined in section
+[Calculation of Transaction Type AA_TX_TYPE hash](#calculation-of-transaction-type-aatxtype-hash).\
+The `version` parameter is added in order to maintain the Solidity method ID in case of changes to this struct
+in future revisions of this EIP.
+
+The amount of gas used by this frame is referred to as `senderValidationGasUsed`.
+
+The frame must return 32 bytes `validationData` that is interpreted as:
+
+```solidity
+
+abi.encodePacked(MAGIC_VALUE_SENDER, validBefore, validAfter)
+
+```
+
+In order to allow a gas estimation to determine the amount of gas that this frame
+requires to complete successfully while not having the actual `signature` value, this function
+should avoid reverting on invalid signature, and should return a value different from `MAGIC_VALUE_SENDER`.
```

Comment:
Should this also specify that the function should continue its execution until the end on an invalid signature to allow for a correct gas estimation?

Reply:
Yes, it should consume the same amount of gas as it would with a correct signature. It might not even be just to continue execution, but actually to use the same amount of gas. For example, a multisig that needs to validate 3 signatures and received none, fails the signature check immediately but should waste the gas of validating three signatures.

Reply:
the wording should be that it MUST return the `MAGIC_VALUE` to mark success, and SHOULD continue execution in signature error.